client/object_put.go

```diff
@@ -0,0 +1,198 @@
+package client
+
+import (
+	"context"
+	"crypto/ecdsa"
+	"fmt"
+
+	v2object "github.com/nspcc-dev/neofs-api-go/v2/object"
+	rpcapi "github.com/nspcc-dev/neofs-api-go/v2/rpc"
+	"github.com/nspcc-dev/neofs-api-go/v2/rpc/client"
+	v2session "github.com/nspcc-dev/neofs-api-go/v2/session"
+	"github.com/nspcc-dev/neofs-sdk-go/object"
+	oid "github.com/nspcc-dev/neofs-sdk-go/object/id"
+	"github.com/nspcc-dev/neofs-sdk-go/session"
+	"github.com/nspcc-dev/neofs-sdk-go/token"
+)
+
+// PrmObjectPutInit groups parameters of ObjectPutInit operation.
+//
+// At the moment the operation is not parameterized, however,
+// the structure is still declared for backward compatibility.
+type PrmObjectPutInit struct{}
+
+// ResObjectPut groups the final result values of ObjectPutInit operation.
+type ResObjectPut struct {
+	statusRes
+
+	resp v2object.PutResponse
+}
+
+// ReadStoredObject reads identifier of the saved object.
+// Returns false if ID is missing (not read).
+func (x *ResObjectPut) ReadStoredObject(id *oid.ID) bool {
```

Comment:
Function reads ID, not the object.

Also, if we want to use non pointer ID values (as in params), I don't see much sense in having function that takes pointer as an argument. Using non pointer values means that we do not reuse them. So I would expect simple getter:
```go
func (x *ResObjectPut) ObjectID() (oid.ID, error)
```
or even
```go
func (x *ResObjectPut) ObjectID() oid.ID
```

with empty object ID if, for some reason, it was missed in the response.

Reply:
I'll add `ID` suffix to the name.

Proposed method signature is the most flexible IMO because it allows to reuse already allocated ID variables.